controllers/spacebindingcleanup/spacebinding_cleanup_controller.go

```diff
@@ -94,8 +96,71 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 
 func (r *Reconciler) deleteSpaceBinding(logger logr.Logger, spaceBinding *toolchainv1alpha1.SpaceBinding) error {
 	logger.Info("deleting the SpaceBinding")
+
+	// check if spaceBinding was created from SpaceBindingRequest,
+	// in that case we must delete the SBR and will take care of deleting the SpaceBinding as well
+	spaceBindingReqeuestAssociated := checkSpaceBindingRequestAssociated(spaceBinding)
+	if spaceBindingReqeuestAssociated.found {
+		return r.deleteSpaceBindingRequest(logger, spaceBindingReqeuestAssociated)
```

Comment:
Let's also check if the SBR functionality is enabled in ToolchainConfig - if it's disabled but there is still SB created via SBR, then I would return an error.

Reply:
I was thinking at the scenario in which for some reason we want to disable the SBR functionality (maybe temporarily), if we do this the cleanup of the SBRs will not be performed automatically and a bunch of errors will pop up in our logs.

Maybe it's enough to just log the error (a warning would be more suited IMO)  but still perform the cleanup ? WDYT ?  

Just wanted to better understand why we should not clean them up.

Reply:
> If the SBR functionality is disabled, then the SBR controller is not running, which means that the SBR CR will stay in a Terminating state.

Good point! If we restarted the operator then yes the controller will not be running.

> How about doing both - calling the delete, but at the same time either returning or logging an error?
I think this would be easier to implement, because I "just" need to return an error after delete if the config is disabled. Also it covers the case in which we disabled the config by mistake or there was some issue with config and was unintentionally turned off without restarting the operator, the cleanup will still be done by the controller.

Reply:
ok, let's do that :+1:

Reply:
Actually logging the error was easier than returning it, done in https://github.com/codeready-toolchain/host-operator/pull/857/commits/98af28b3ecdfb02ba2c883facc2331982b8c3258

I can return it, but I thought there was no additional value in doing that compared to just logging it, since the resources will be reconciled anyway once we restart the operators (and once every N hours if I'm not mistaken). 
But please let me know if I missed anything and we should return it instead. 

Please also check new PR for unit testing utilities: https://github.com/codeready-toolchain/toolchain-common/pull/323